src/nanotron/optim/named_optimizer.py

```diff
@@ -12,6 +12,7 @@ def __init__(
         self,
         named_params_or_groups: Iterable[Union[Tuple[str, torch.Tensor], Dict[str, Any]]],
         optimizer_builder: Callable[[Iterable[Dict[str, Any]]], torch.optim.Optimizer],
+        weight_decay: float = 0.0,
```

Comment:
it looks like we don't use weight_decay? i searched the `weight_decay` argument in the NamedOptimizer's __init__, it doens't use this argument?